fsspec/registry.py

```diff
@@ -154,6 +154,10 @@ def register_implementation(name, cls, clobber=False, errtxt=None):
         "class": "fsspec.implementations.git.GitFileSystem",
         "err": "Install pygit2 to browse local git repos",
     },
+    "smb": {
```

Comment:
Is there a distinction to be made between SMB protocols here?

Reply:
I have found only `smb://`  in [SMB Filesharing URL Scheme](https://tools.ietf.org/html/draft-crhertel-smb-url-00#section-2).

Do you know something else?

Reply:
I have no idea, samba is alien to me :|
If your implementation tries to connect to a service of the other protocol, you just fail, right?
Perhaps @telamonian has an opinion.

Reply:
I need to look up some specs/how `pysmb` does it, but I think it's essentially correct to have the single protocol. You can then have kwargs that limit you to eg no SMB1. I'll look up the details

Reply:
Okay, so this isn't covered by the standard (it instead describes [a flexible negotiation process](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-cifs/80850595-e301-4464-9745-58e4945eb99b) for which features a connection will support; this is why smb has a reputation of being security swiss-cheese).

On the other hand, `smbprotocol` itself provides a nice and simple way of [specifying a minimum SMB_X protocol for your connection](https://github.com/jborean93/smbprotocol#additional-info):

>By default, the negotiation process will use the latest dialect that is supported by the server but this can be overridden if required. When this is done by the following code
>
>```
>import uuid
>
>from smbprotocol.connection import Connection, Dialects
>
>connection = Connection(uuid.uuid4(), "server", 445)
>connection.connect(Dialects.SMB_3_0_2)
>```
>
>While you shouldn't want to downgrade to an earlier version, this does allow you to set a minimum dialect version if required.

Ideally we should support a `dialect` kwarg for `SMBFileSystem`, which would take a [`Dialects`](https://github.com/jborean93/smbprotocol/blob/40161522e3add7ec0382c08042b343d012c1b0ec/smbprotocol/__init__.py#L49) arg. Possibly it would be simpler for `dialect` to instead take a `str` arg, which would then get parsed to a `Dialects`, since this would prevent endusers from having to do `from smbprotocol.connection import Dialects` in their own code.

Reply:
Does that imply that smbprotocol does, in theory, support all protocols? Then this really is an ideal solution, better than I had original thought.
Agree that being to set parameters on the `connect` call is useful. If they are few, they could have specific entries in `__init__`, or we could have a `connect_kwargs=` if there are many possibilities.

Reply:
So, roughly speaking it looks like `pysmb` supports SMB1 and SMB2, while `smbprotocol` supports SMB2 and SMB3 (with `smbprotocol` having more granular support for specific SMB_X_Y_Z versions).

I'm personally more excited by the fact that `smbprotocol` actually has proper Windows CI set up for their repo (my enthusiasm is slightly dampened by the fact [said CI is currently failing](https://ci.appveyor.com/project/jborean93/smbprotocol/builds/33450600); though it looks like it's specifically a python 3.5 issue. Which I don't care about, and nor should anyone else).

Reply:
We *do not* care about py35 :)

Reply:
How does SMB1/2/3 map to usage in the wild?

Reply:
If you're on Windows 7 or higher, then you can use SMB2. So there's allllmost no extant use case for SMB1, which is good, because it is super insecure

Reply:
Win7 is past end-of-life, I say we ignore it; "smb" will mean this implementation only. In the unlikely case that someone makes an SMB1 implementation (like the pyfilesystems one), then we can give it a different protocol marker.

>  September 17, 2019 | 4.11 | SMB1 is disabled by default.

Reply:
> Win7 is past end-of-life

LOL you wish. You might be shocked at the millions upon millions of remaining corporate and government users. I know I was.

But it shouldn't be an issue here, since Win7 does support SMB2. So I agree with this being the implementation for `smb://`.

Reply:
OK :)